app/src/main/java/com/babylon/wallet/android/presentation/account/composable/MetadataView.kt

```diff
@@ -36,117 +43,171 @@ import java.time.Instant
 import java.time.ZoneId
 import java.time.format.DateTimeFormatter
 import java.time.format.FormatStyle
-import java.util.Locale
 
 @Composable
-fun AssetMetadataRow(
-    modifier: Modifier,
+fun MetadataView(
+    modifier: Modifier = Modifier,
     key: String,
-    valueView: @Composable RowScope.() -> Unit
+    isLocked: Boolean = false,
+    isRenderedInNewLine: Boolean = false,
+    valueView: @Composable () -> Unit
 ) {
-    Row(
-        modifier = modifier,
-        horizontalArrangement = Arrangement.SpaceBetween,
-        verticalAlignment = Alignment.CenterVertically
-    ) {
-        Text(
-            modifier = Modifier.padding(end = RadixTheme.dimensions.paddingMedium),
-            text = key.replaceFirstChar {
-                if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
-            },
-            style = RadixTheme.typography.body1Regular,
-            color = RadixTheme.colors.gray2
-        )
-
-        valueView()
-    }
-}
-
-@Composable
-fun Metadata.View(modifier: Modifier) {
     if (isRenderedInNewLine) {
         Column(
             modifier = modifier,
             verticalArrangement = Arrangement.spacedBy(RadixTheme.dimensions.paddingSmall)
         ) {
-            KeyView()
-            ValueView(isRenderedInNewLine = true)
+            MetadataKeyView(key = key, isLocked = isLocked)
+            valueView()
         }
     } else {
         Row(
             modifier = modifier,
             horizontalArrangement = Arrangement.SpaceBetween,
-            verticalAlignment = Alignment.CenterVertically
+            verticalAlignment = Alignment.Top
         ) {
-            KeyView()
+            MetadataKeyView(key = key, isLocked = isLocked)
```

Comment:
Maybe it is worth to allocate some max width to the metadata key, I noticed that for some metadata names, long ones, there is not much space left for the value. When increasing font size a bit in the system (from already quite big default setting on samsung), value is pushed out of the screen

Reply:
@jakub-rdx I pushed a commit ditching the `Row` and replacing it with constraint layout. It works better know. If a key has more than 30 chars it divides the space a little better.